DANS-KNAW/dd-manage-deposit
# DD-1336 Finish-dd-manage-deposit

Fixes DD-1336 Finish-dd-manage-deposit

# Description of changes
- Added finalised column names to the `deposit_properties` table
- Changed the code basis on new columns (DepositProperties class)
- The absolute path as location of deposits couldn't resolve in the server side completely and/or correctly. The functions parameter `Path` are substituted with `File`
- The string limits defined for `location` and `description` columns + code changes for these
- `poolingInterval` is now configurable via service's configuration file + code changes for this 
- `deleted` column is protected now and delete notification avoided during moving of the deposit folders
- Adapted the rest of code according the changes

# How to test
- Virtaul box -vagrant - deploy (dd-manage-depost) :  `https://dar.dans.knaw.nl/dataverse/root/`
- add SWORD2 zipped bags:
**Ex.**. `curl -v -i -X POST -H "Content-Disposition: attachment; filename= test-bag-1.zip" -H 'Content-Type: application/zip' -H "Content-MD5: $(md5 -q test-bag-1.zip)" -H 'Packaging: http://purl.org/net/sword/package/BagIt' --data-binary "@test-bag-1.zip" -u user001:user001 http://dar.dans.knaw.nl:20320/collection/1`

- look at / follow the database `report` via:
  - on browser: `http://dar.dans.knaw.nl:20355/report`
  -  as json records: `curl -X GET http://dar.dans.knaw.nl:20355/report`
  - as csv text: `curl -X GET -H 'Accept: text/csv' http://dar.dans.knaw.nl:20355/report`
- look at the folders via vagrant ssh: `tree -L 3 /var/opt/dans.knaw.nl/tmp/auto-ingest/`
- change `deposits.properties` using `nano` ==> the records should be refreshed (specially `depositUpdateTimestamp`) 
- move the deposits using `mv` command ==> `location` column should show the last deposit path
- delete the deposits using `rm -fr` command ==> `deleted` column should changed from `false` into `true`
- delete records using filters, such as:
  - `curl -X POST http://dar.dans.knaw.nl:20355/delete-deposit\?deleted\=true`
  - `curl -X POST http://dar.dans.knaw.nl:20355/delete-deposit\?user=user001` 


# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans
